cranelift-codegen/src/ir/function.rs

```diff
@@ -197,10 +205,12 @@ impl Function {
             !self.offsets.is_empty(),
             "Code layout must be computed first"
         );
+        let mut divert = RegDiversions::new();
+        divert.at_ebb(&self.entry_diversions, ebb);
```

Comment:
Maybe make a wrapper `RegDiversions::new_at()`?

Reply:
Other parts of the code have these separated in 2, a `RegDiversions::new()` out-side a loop, and the `at_ebb` call within the loop. I think this code is more idiomatic if it remains separated in 2 statements.